storm-core/src/jvm/org/apache/storm/ui/UIHelpers.java

```diff
@@ -0,0 +1,319 @@
+import org.eclipse.jetty.server.ssl.SslSocketConnector;
+import org.eclipse.jetty.servlet.FilterHolder;
+import org.eclipse.jetty.servlet.FilterMapping;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.servlets.CrossOriginFilter;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.json.simple.JSONValue;
+
+import javax.servlet.Servlet;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.net.URLEncoder;
+import java.util.*;
+
+public class UIHelpers {
+
+    private static final String[][] PRETTY_SEC_DIVIDERS = {
+            new String[]{"s", "60"},
+            new String[]{"m", "60"},
+            new String[]{"h", "24"},
+            new String[]{"d", null}};
+
+    private static final String[][] PRETTY_MS_DIVIDERS = {
+            new String[]{"ms", "1000"},
+            new String[]{"s", "60"},
+            new String[]{"m", "60"},
+            new String[]{"h", "24"},
+            new String[]{"d", null}};
+
+    public static String prettyUptimeStr(String val, String[][] dividers) {
+        int uptime = Integer.parseInt(val);
+        LinkedList<String> tmp = new LinkedList<>();
+        for (String[] divider : dividers) {
+            if (uptime > 0) {
+                if (divider[1] != null) {
+                    int div = Integer.parseInt(divider[1]);
+                    tmp.addFirst(uptime % div + divider[0]);
+                    uptime = uptime / div;
+                } else {
+                    tmp.addFirst(uptime + divider[0]);
+                }
+            }
+        }
+        return Joiner.on(" ").join(tmp);
+    }
+
+    public static String prettyUptimeSec(String sec) {
+        return prettyUptimeStr(sec, PRETTY_SEC_DIVIDERS);
+    }
+
+    public static String prettyUptimeSec(int secs) {
+        return prettyUptimeStr(String.valueOf(secs), PRETTY_SEC_DIVIDERS);
+    }
+
+    public static String prettyUptimeMs(String ms) {
+        return prettyUptimeStr(ms, PRETTY_MS_DIVIDERS);
+    }
+
+    public static String prettyUptimeMs(int ms) {
+        return prettyUptimeStr(String.valueOf(ms), PRETTY_MS_DIVIDERS);
+    }
+
+
+    public static String urlFormat(String fmt, Object... args) {
+        String[] argsEncoded = new String[args.length];
+        for (int i = 0; i < args.length; i++) {
+            argsEncoded[i] = URLEncoder.encode(String.valueOf(args[i]));
+        }
+        return String.format(fmt, argsEncoded);
+    }
+
+    public static String prettyExecutorInfo(ExecutorInfo e) {
+        return "[" + e.get_task_start() + "-" + e.get_task_end() + "]";
+    }
+
+    public static Map unauthorizedUserJson(String user) {
+        return ImmutableMap.of(
+                "error", "No Authorization",
+                "errorMessage", String.format("User %s is not authorized.", user)
+        );
+    }
+
+    public static List unauthorizedUserHtml(String user) {
```

Comment:
This is very hiccup specific and is now only used by logviewer.  I would prefer to see this removed and the original function placed in logviewer so when we redo the logviewer code and replace hiccup we can remove this code entirely.